AzureDevOpsPowerShell/Public/Api/Git/PullRequests/New-AzDoPullRequest.ps1

```diff
@@ -0,0 +1,154 @@
+        RepoName         = "Repo1"
+        ProjectName   = "Project 1"
+        Title          = "New Pull Request"
+        Description    = "This is a new pull request"
+        SourceRefName  = "refs/heads/feature1"
+        TargetRefName  = "refs/heads/main"
+    }
+    $params | New-AzDoPullRequest
+
+    This example creates a new Azure DevOps Pull Request with pipeline parameters
+.OUTPUTS
+    [PSCustomObject]@{
+        CollectionUri  = $CollectionUri
+        ProjectName    = $ProjectName
+        RepoName       = $RepoName
+        PullRequestId  = $res.pullRequestId
+        PullRequestURL = $res.url
+      }
+.NOTES
+#>
+
+  [CmdletBinding(SupportsShouldProcess)]
+  param (
+    # Collection Uri of the organization
+    [Parameter(Mandatory)]
+    [ValidateScript({ Validate-CollectionUri -CollectionUri $_ })]
+    [string]
+    $CollectionUri,
+
+    # Id of the repository
+    [Parameter(Mandatory)]
+    [string]
+    $RepoName,
+
+    # Name of the project where the new repository has to be created
+    [Parameter(Mandatory)]
+    [string]
+    $ProjectName,
+
+    # Title of the pull request
+    [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+    [string[]]
+    $Title,
+
+    # Description of the pull request
+    [Parameter(ValueFromPipelineByPropertyName)]
+    [string[]]
+    $Description = 'Describe the changes made in this pull request',
+
+    # Source ref name
+    [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+    [string[]]
+    $SourceRefName,
+
+    # Target ref name
+    [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+    [string[]]
+    $TargetRefName
+  )
+
+  begin {
+    Write-Verbose "Starting function: New-AzDoPullRequest"
+    $CollectionUri = $CollectionUri.TrimEnd('/')
+    $result = New-Object System.Collections.Generic.List[System.Object]
+  }
+
+  process {
+    foreach ($pr in $Title) {
+      $prTitle = $pr
+      $prDescription = $Description[$Title.IndexOf($pr)]
+      $prSourceRefName = $SourceRefName[$Title.IndexOf($pr)]
+      $prTargetRefName = $TargetRefName[$Title.IndexOf($pr)]
+
+      # If the ref name is not in the format refs/heads/branch, then add it
+      if ($prSourceRefName -notlike "refs/*") {
+        $prSourceRefName = "refs/heads/$prSourceRefName"
+      }
+      if ($prTargetRefName -notlike "refs/*") {
+        $prTargetRefName = "refs/heads/$prTargetRefName"
+      }
+
+      $params = @{
+        uri     = "$CollectionUri/$ProjectName/_apis/git/repositories/$RepoName/pullrequests"
+        version = '7.2-preview.2'
+        method  = 'POST'
+        body    = @{
+          sourceRefName = $prSourceRefName
+          targetRefName = $prTargetRefName
+          title         = $prTitle
+          description   = $prDescription
+        }
+      }
+
+      if ($PSCmdlet.ShouldProcess($CollectionUri, "Create pull request named: $($PSStyle.Bold)$prTitle$($PSStyle.Reset)")) {
+        try {
+          $result += Invoke-AzDoRestMethod @params
```

Comment:
@Dylan-Prins could you please check this, had to create a dynamic array for this to work